Add more wake lock tests

Honry opened the pull request:
- test in an insecure context
  - test for invoking 'cancel()' twice
  - test when document is hidden
  - test for applicability when screen is switched off

https://github.com/w3c/wake-lock/issues/111

<!-- Reviewable:start -->

<!-- Reviewable:end -->

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20605)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/302262055)

marcoscaceres commented on wake-lock/wakelock-applicability-manual.https.html:
```diff
@@ -0,0 +1,48 @@
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<p>
+  Lock and turn off the screen, then turn it on and unlock the screen in a minute.
```
This is unclear, are you asking the tested to wait 60 seconds?

marcoscaceres replied:
I theory, the tests will run by the time the tested turns the screen off and then on again. So likely they only have to wait maybe 1 or 2 seconds, right?

Honry replied:
I didn't mean waiting, I just wanted to make sure the interaction should be complete within 60 seconds, before test timeout. Anyway, I will improve the description.

marcoscaceres commented on wake-lock/wakelock-applicability-manual.https.html:
```diff
@@ -0,0 +1,48 @@
+<title>wake lock applicability test</title>
+<link rel="help" href="https://w3c.github.io/wake-lock/#dfn-the-wake-lock-is-applicable">
+<meta name="flags" content="interact">
+<script src=/resources/testharness.js></script>
```
nit: unquoted attributes.

marcoscaceres commented on wake-lock/wakelock-applicability-manual.https.html:
```diff
@@ -0,0 +1,48 @@
+  assert_false(document.hidden, "document is visiable when screen is unlocked");
+  assert_true(wakeLock.active, "the screen wake lock is active when screen is switched on again");
+  request.cancel();
+}, "The screen wake lock isn't applicable after the screen is manually swiched off"
```
Nit: "applicable" is a somewhat confusing term. Maybe "usable"? dunno.

Honry replied:
Hmm.. This sentence is referenced from [spec](https://w3c.github.io/wake-lock/#dfn-the-wake-lock-is-applicable), I think we can leave it as is.

> The screen wake lock MUST NOT be applicable after the screen is manually swiched off by the user until it is switched on again. Manually switching off the screen MUST NOT affect the applicability of the system wake lock.

marcoscaceres replied:
ah, thanks for clarifying... ok, yeah, it makes sense in the context of the spec.

marcoscaceres commented on wake-lock/wakelock-cancel-twice.https.html:
```diff
@@ -0,0 +1,20 @@
+  //If the cancel() method has already been invoked on this object,
+  //abort these steps, no error fired
+  request.cancel();
+}, document.title);
```
Cute hack :) However, if we end up adding more tests to this file, people looking at the test might get annoyed if they have to scroll to the top to find out what the test does. Just inline the testable assertion here and maybe generalize the doc's title.

Honry replied:
:) Will fix.

marcoscaceres commented on wake-lock/wakelock-insecure-context.html:
```diff
@@ -0,0 +1,13 @@
+
+test(t => {
+  assert_false("getWakeLock" in navigator, "'getWakeLock' must not be exposed");
+}, document.title);
```
as above.

marcoscaceres reviewed (approved):
Couple of nits, but these are looking great @Honry! 🏆 

I'd like to get a review from @andrey-logvinov before we merge. @andrey-logvinov, can you please double check that the algorithms are being fully tested?

Honry committed:
Add more wake lock tests

  - test in an insecure context
  - test for invoking 'cancel()' twice
  - test when document is hidden
  - test for applicability when screen is switched off

Honry committed:
Fix trailing whitespace

Honry committed:
Fix nits

Honry commented:
> I'd like to get a review from @andrey-logvinov before we merge. @andrey-logvinov, can you please double check that the algorithms are being fully tested?

@andrey-logvinov, friendly ping. :)

marcoscaceres commented:
Merging, as it's been quite a few days - @andrey-logvinov, could you please still have a look at this and file any bugs as appropriate?
